build-tools-internal/build.gradle

```diff
@@ -119,6 +119,10 @@ gradlePlugin {
       id = 'elasticsearch.internal-java-rest-test'
       implementationClass = 'org.elasticsearch.gradle.internal.test.rest.InternalJavaRestTestPlugin'
     }
+    newJavaRestTest {
```

Comment:
I don't have strong feelings on making this name the new default and renaming all the old vs the current convention and renaming after we've prune the old stuff. I named it "new" because it's not obvious how long it'll take to migrate everything.

I agree, renaming the current plugins to "legacy" might make it obvious they are "old" and "deprecated".

Reply:
+1 for renaming current plugins to "legacy". Calling something "new" somehow implies to me its not the standard way (yet) to do things but it should be IMO

Reply:
I've renamed the old plugins using a "legacy" prefix.

`internal-yaml-rest-test` -> `legacy-yaml-rest-test`
`internal-java-rest-test` -> `legacy-java-rest-test`
`yaml-rest-compat-test` -> `legacy-yaml-rest-compat-test`